libethereum/Executive.cpp

```diff
@@ -306,19 +306,41 @@ bool Executive::call(CallParameters const& _p, u256 const& _gasPrice, Address co
 	return !m_ext;
 }
 
-bool Executive::create(Address _sender, u256 _endowment, u256 _gasPrice, u256 _gas, bytesConstRef _init, Address _origin)
+bool Executive::create(Address _txSender, u256 _endowment, u256 _gasPrice, u256 _gas, bytesConstRef _init, Address _origin)
 {
-	u256 nonce = m_s.getNonce(_sender);
-	if (_sender != MaxAddress) // EIP86
-		m_s.incNonce(_sender);
+	if (m_envInfo.number() >= m_sealEngine.chainParams().u256Param("metropolisForkBlock"))
+	{
+		m_newAddress = right160(sha3(MaxAddress.asBytes() + sha3(_init).asBytes()));
+		return executeCreate(_txSender, _endowment, _gasPrice, _gas, _init, _origin);
+	}
+	else
+		return createOpcode(_txSender, _endowment, _gasPrice, _gas, _init, _origin);
+}
+
+bool Executive::createOpcode(Address _txSender, u256 _endowment, u256 _gasPrice, u256 _gas, bytesConstRef _init, Address _origin)
+{
+	u256 nonce = m_s.getNonce(_txSender);
+	if (_txSender != MaxAddress) // EIP86
+		m_s.incNonce(_txSender);
```

Comment:
So now the nonce can be incremented only for CREATE and for create tx before Metropolis, but neither for CREATE_P2SH nor for create tx after Metropolis. Is this correct?